### ISSUE TITLE
Replies broke

### ISSUE THREAD

**Issue by Fabian42** [TediCross/TediCross#119]
Replies do not work anymore. They appear as simple messages on Discord.

**Comment by Suppen** [comment]
I cannot reproduce this. In my test groups, replies work as intended. Can you give a list of instructions to reproduce the error?

**Comment by Fabian42** [comment]
It's the same setup as before, just a bot update must have broken it. Steps are as usual, reply to a message on Telegram (written by yourself or someone else) and it should quote and ping the user on Discord. But it doesn't.

Here are the relevant parts of my settings:
```
telegram:
  useFirstNameInsteadOfUsername: false
  colonAfterSenderName: false
  skipOldMessages: false
  sendEmojiWithStickers: true
  token: '00000'
discord:
  skipOldMessages: false
  useNickname: true
  displayTelegramReplies: inline
  replyLength: 100
  maxReplyLines: 2
  token: 00000
debug: false
bridges:
…
  - name: sl_general
    direction: both
    telegram:
      chatId: -00000
      relayJoinMessages: true
      relayLeaveMessages: true
      sendUsernames: false
      ignoreCommands: false
    discord:
      serverId: '00000'
      channelId: '00000'
      relayJoinMessages: true
      relayLeaveMessages: true
      sendUsernames: true
      crossDeleteOnTelegram: false
…
```

**Comment by Suppen** [comment]
Which version are you running? Which version was the last it worked in?

**Comment by Fabian42** [comment]
I updated in case it was fixed already and it still happens. I don't know which version it broke in, I just noticed it now. I don't usually look at the Discord side. I definitely replied successfully on 2019-06-07, but I might have used an old version at that point.

**Comment by Suppen** [comment]
This is a test from my test channels, which runs 0.9.5. I have also confirmed this is the behavior on the bot running TediCross support channel, which is also 0.9.5. I have tested with replies set to both inline and embedded. I cannot reproduce the problem.
![TediCross](https://user-images.githubusercontent.com/12338293/62491152-758e1280-b7cb-11e9-89d5-eb88e0fd5fd9.png)

Anything in the console? Any specific users it doesn't work for? Any specific types of messages? Or just any replies?

**Comment by Fabian42** [comment]
All replies, I just tried regular text. Nothing in the console.

**Comment by Suppen** [comment]
Allright. Let's try some debugging. It involves some simple code changes.
In the file `src/telegram2discord/middlewares.js`, after line 331, can you add the following line of code?
```
console.log(ctx.tediCross);
```
A snippet of the file should then look like this:
```
                  ...
                  // Turn the original text into "<no text>" if there is no text
                  if (R.isEmpty(ctx.tediCross.replyTo.text.raw)) {
                          ctx.tediCross.replyTo.text.raw = "<no text>";
                  }
>>                console.log(ctx.tediCross);
          }
  
          next();
  }
...
```

Then restart TediCross, try to reply to something, and send me the output from the console

**Comment by Fabian42** [comment]
Aug  8 13:08:48 raspberrypi tedicross[285]: { message:
Aug  8 13:08:48 raspberrypi tedicross[285]:    { message_id: 10133,
Aug  8 13:08:48 raspberrypi tedicross[285]:      from:
Aug  8 13:08:48 raspberrypi tedicross[285]:       { id: 21785454,
Aug  8 13:08:48 raspberrypi tedicross[285]:         is_bot: false,
Aug  8 13:08:48 raspberrypi tedicross[285]:         first_name: 'Fabian',
Aug  8 13:08:48 raspberrypi tedicross[285]:         last_name: 'Röling',
Aug  8 13:08:48 raspberrypi tedicross[285]:         username: 'Fabian42' },
Aug  8 13:08:48 raspberrypi tedicross[285]:      chat:
Aug  8 13:08:48 raspberrypi tedicross[285]:       { id: -1001209285859,
Aug  8 13:08:48 raspberrypi tedicross[285]:         title: 'sl_general',
Aug  8 13:08:48 raspberrypi tedicross[285]:         username: 'sl_general',
Aug  8 13:08:48 raspberrypi tedicross[285]:         type: 'supergroup' },
Aug  8 13:08:48 raspberrypi tedicross[285]:      date: 1565262528,
Aug  8 13:08:48 raspberrypi tedicross[285]:      reply_to_message:
Aug  8 13:08:48 raspberrypi tedicross[285]:       { message_id: 10132,
Aug  8 13:08:48 raspberrypi tedicross[285]:         from: [Object],
Aug  8 13:08:48 raspberrypi tedicross[285]:         chat: [Object],
Aug  8 13:08:48 raspberrypi tedicross[285]:         date: 1565262519,
Aug  8 13:08:48 raspberrypi tedicross[285]:         text: "Fabian42\nBoo,
I'm a human now!",
Aug  8 13:08:48 raspberrypi tedicross[285]:         entities: [Array] },
Aug  8 13:08:48 raspberrypi tedicross[285]:      text: "And now I'm a bot
trying to reply." },
Aug  8 13:08:48 raspberrypi tedicross[285]:   messageId: 10133,
Aug  8 13:08:48 raspberrypi tedicross[285]:   bridges:
Aug  8 13:08:48 raspberrypi tedicross[285]:    [ Bridge {
Aug  8 13:08:48 raspberrypi tedicross[285]:        name: 'sl_general',
Aug  8 13:08:48 raspberrypi tedicross[285]:        direction: 'both',
Aug  8 13:08:48 raspberrypi tedicross[285]:        telegram:
[BridgeSettingsTelegram],
Aug  8 13:08:48 raspberrypi tedicross[285]:        discord:
[BridgeSettingsDiscord] } ],
Aug  8 13:08:48 raspberrypi tedicross[285]:   from:
Aug  8 13:08:48 raspberrypi tedicross[285]:    { firstName: 'Fabian',
lastName: 'Röling', username: 'Fabian42' },
Aug  8 13:08:48 raspberrypi tedicross[285]:   replyTo:
Aug  8 13:08:48 raspberrypi tedicross[285]:    { isReplyToTediCross: true,
Aug  8 13:08:48 raspberrypi tedicross[285]:      message:
Aug  8 13:08:48 raspberrypi tedicross[285]:       { message_id: 10132,
Aug  8 13:08:48 raspberrypi tedicross[285]:         from: [Object],
Aug  8 13:08:48 raspberrypi tedicross[285]:         chat: [Object],
Aug  8 13:08:48 raspberrypi tedicross[285]:         date: 1565262519,
Aug  8 13:08:48 raspberrypi tedicross[285]:         text: "Fabian42\nBoo,
I'm a human now!",
Aug  8 13:08:48 raspberrypi tedicross[285]:         entities: [Array] },
Aug  8 13:08:48 raspberrypi tedicross[285]:      originalFrom:
Aug  8 13:08:48 raspberrypi tedicross[285]:       { firstName: 'Discord',
Aug  8 13:08:48 raspberrypi tedicross[285]:         lastName: undefined,
Aug  8 13:08:48 raspberrypi tedicross[285]:         username:
'discord001bot' },
Aug  8 13:08:48 raspberrypi tedicross[285]:      text: { raw: "Boo, I'm a
human now!", entities: [] },
Aug  8 13:08:48 raspberrypi tedicross[285]:      dcUsername: 'Fabian42' } }

**Comment by Suppen** [comment]
So it does indeed get and process the reply correctly. Please remove the line again, and then add the following under line 563 in the same file:
```
console.log(ctx.tediCross.prepared);
```
so it looks like
```
...
        )(tc.bridges);
console.log(ctx.tediCross.prepared);
  
        next();
...
```

**Comment by Fabian42** [comment]
Aug  8 18:49:24 raspberrypi tedicross[294]: [ { bridge:
Aug  8 18:49:24 raspberrypi tedicross[294]:      Bridge {
Aug  8 18:49:24 raspberrypi tedicross[294]:        name: 'sl_general',
Aug  8 18:49:24 raspberrypi tedicross[294]:        direction: 'both',
Aug  8 18:49:24 raspberrypi tedicross[294]:        telegram:
[BridgeSettingsTelegram],
Aug  8 18:49:24 raspberrypi tedicross[294]:        discord:
[BridgeSettingsDiscord] },
Aug  8 18:49:24 raspberrypi tedicross[294]:     header: '',
Aug  8 18:49:24 raspberrypi tedicross[294]:     embed: undefined,
Aug  8 18:49:24 raspberrypi tedicross[294]:     file: undefined,
Aug  8 18:49:24 raspberrypi tedicross[294]:     text: "And I'm creating log
files." } ]

**Comment by Suppen** [comment]
Do you have `telegram.sendUsernames` set to `false` on the bridge(s) it's not working on?

**Comment by Suppen** [comment]
Yes, you do. I realised you posted your settings further up.

This causes the header to be dropped. The header is what contains the replied-to text when replies are set to `inline`. I'll see what I can do about that

**Comment by Fabian42** [comment]
bump

**Comment by Suppen** [comment]
Should be fixed in version 0.9.6
